core/include/vecmem/memory/impl/device_atomic_ref.ipp

```diff
@@ -116,137 +187,216 @@ VECMEM_HOST_AND_DEVICE auto device_atomic_ref<T, address>::load(
     return value;
 #elif defined(CL_SYCL_LANGUAGE_VERSION) || defined(SYCL_LANGUAGE_VERSION)
     return __VECMEM_SYCL_ATOMIC_CALL0(load, m_ptr);
+#elif defined(VECMEM_HAVE_BUILTIN_ATOMIC_LOAD) && \
+    defined(VECMEM_HAVE_MEMORDER_DEFINITIONS)
+    return __atomic_load_n(m_ptr, __memorder_vecmem_to_builtin(order));
+#elif defined(__clang__) || defined(__GNUC__) || defined(__GNUG__) || \
+    defined(__CUDACC__)
+    return __atomic_load_n(m_ptr, __memorder_vecmem_to_builtin(order));
+#elif defined(_MSC_VER)
+    // Probably pollutes your caches but who cares about MSVC anyway.
+    return InterlockedAdd(m_ptr, 0);
```

Comment:
The following would probably behave better:

```suggestion
    return InterlockedOr(m_ptr, static_cast<T>(0));
```

Reply:
Sure, that will work. :slightly_smiling_face: